aws/table_aws_api_gatewayv2_integration.go

```diff
@@ -0,0 +1,274 @@
+			},
+			{
+				Name:        "template_selection_expression",
+				Description: "The template selection expression for the integration. Supported only for WebSocket APIs.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "tls_config",
+				Description: "The TLS configuration for a private integration. If you specify a TLS configuration, private integration traffic uses the HTTPS protocol. Supported only for HTTP APIs.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard columns
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAPIGatewayV2IntegrationAkas,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("IntegrationId"),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAPIGatewayV2Integrations(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAPIGatewayV2Integrations", "AWS_REGION", region)
+	api := h.Item.(*apigatewayv2.Api)
+
+	// Create Session
+	svc, err := APIGatewayV2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	pagesLeft := true
+	params := &apigatewayv2.GetIntegrationsInput{
+		ApiId: api.ApiId,
+	}
+
+	for pagesLeft {
+		result, err := svc.GetIntegrations(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, integration := range result.Items {
+			d.StreamLeafListItem(ctx, integrationInfo{*integration, *api.ApiId})
+		}
+
+		if result.NextToken != nil {
+			pagesLeft = true
+			params.NextToken = result.NextToken
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAPIGatewayV2Integration(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAPIGatewayV2Integration")
+	api := d.KeyColumnQuals["api_id"].GetStringValue()
+	key := d.KeyColumnQuals["integration_id"].GetStringValue()
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	// Create Session
+	svc, err := APIGatewayV2Service(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &apigatewayv2.GetIntegrationInput{
+		ApiId:         aws.String(api),
+		IntegrationId: aws.String(key),
+	}
+
+	item, err := svc.GetIntegration(params)
+
+	if err != nil {
+		plugin.Logger(ctx).Debug("getAPIGatewayV2API__", "ERROR", err)
+		return nil, err
+	}
+
+	if item != nil {
+		integration := &apigatewayv2.Integration{
+			ApiGatewayManaged:                      item.ApiGatewayManaged,
```

Comment:
Why we need this mapping ?

Reply:
We need this since the list and get responses have different structures and we didn't have `api_id` in any of the responses, which was required as part of the feature request.